File: AfterLecture/PlayingWithAndroid/app/src/main/java/no/hiof/larseknu/playingwithandroid/MainActivity.java
package no.hiof.larseknu.playingwithandroid;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    private String LOG_TAG = "PlayingWithAndroid";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Log.d(LOG_TAG, "onCreate() ran");
    }

    @Override
    protected void onStart() {
        super.onStart();

        Log.d(LOG_TAG, "onStart() ran");
    }

    @Override
    protected void onResume() {
        super.onResume();

        Log.d(LOG_TAG, "onResume() ran");
    }

    @Override
    protected void onPause() {
        super.onPause();

        Log.d(LOG_TAG, "onPause() ran");
    }

    @Override
    protected void onStop() {
        super.onStop();

        Log.d(LOG_TAG, "onStop() ran");
    }

    @Override
    protected void onRestart() {
        super.onRestart();

        Log.d(LOG_TAG, "onRestart() ran");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        Log.d(LOG_TAG, "onDestroy() ran");
    }

    public void navigateToOtherActivity(View view) {
        Intent startOtherActivityIntent = new Intent(this, OtherActivity.class);

        startActivity(startOtherActivityIntent);
    }
}

File: Lecture01_Introductions/AfterLectureKotlin/PlayingWithAndroidKotlin/app/src/main/java/no/hiof/larseknu/playingwithandroidkotlin/MainActivity.kt
package no.hiof.larseknu.playingwithandroidkotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View

class MainActivity : AppCompatActivity() {
    private var LOG_TAG = "PlayingWithAndroid";

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onStart() {
        super.onStart()
        Log.d(LOG_TAG, "onStart() ran")
    }

    override fun onResume() {
        super.onResume()
        Log.d(LOG_TAG, "onResume() ran")
    }

    override fun onPause() {
        super.onPause()
        Log.d(LOG_TAG, "onPause() ran")
    }

    override fun onStop() {
        super.onStop()
        Log.d(LOG_TAG, "onStop() ran")
    }

    override fun onRestart() {
        super.onRestart()
        Log.d(LOG_TAG, "onRestart() ran")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(LOG_TAG, "onDestroy() ran")
    }

    fun navigateToOtherActivity(view: View) {
        var intent = Intent(this, OtherActivity::class.java)
        startActivity(intent)
    }
}
